sc_bucket_cleanup/app.py

```diff
@@ -0,0 +1,131 @@
+import boto3
+import logging
+import os
+
+from botocore.exceptions import ClientError
+from datetime import datetime, timedelta, timezone
+
+logging.basicConfig(level=logging.INFO)
+
+MAX_DAYS_FROM_DELETION = 90
+
+def _get_s3_client():
+  return boto3.client('s3')
+
+def _get_cfn_client():
+  return boto3.client('cloudformation')
+
+def _get_env_var_value(env_var):
+  """Get the value of an environment variable
+  :param env_var: The environment variable
+  :returns: The environment variable's value, None if env var is not found
+  """
+  value = os.getenv(env_var)
+  if not value:
+    logging.warning(f'cannot get environment variable: {env_var}')
+
+  return value
+
+def _get_stack_summaries(status_filter):
+  """
+  Get a list of the cloudformation stack summaries applied with a status filter.
+  :param status_filter: The status filter used when listing cloudformation stacks
+  :return: A list of cloudformation stack summaries
+  """
+  cfn = _get_cfn_client()
+  return cfn.list_stacks(StackStatusFilter=status_filter)['StackSummaries']
+
+def _get_stack_resources(stack_name):
+  """
+  Get the information for a cloudformation stack's resource.
+  :param stack_name: The name of the stack or a stack id.  The stack id is
+                     required when getting resources for a deleted stack.
+  :return:
+  """
+  cfn = _get_cfn_client()
+  return cfn.list_stack_resources(StackName=stack_name)['StackResourceSummaries']
+
+def _get_buckets():
+  """
+  Get a list of all buckets in the AWS account
+  :return:
+  """
+  s3 = _get_s3_client()
+  return s3.list_buckets()['Buckets']
+
+def _get_purge_date(days_deleted):
+  """
+  Get the date used to determine whether an AWS resources should be removed.  We
+  inspect any cloudformation stack that was deleted before this date and to see what
+  resource it provisioned.  If that resource still exist in AWS it will be deleted.
+  """
+  return datetime.now(timezone.utc) - timedelta(days=days_deleted)
+
+def _get_deleted_stacks(days_from_deletion):
+  """
+  Get all the deleted Cloudformation stacks that were deployed by the Service
+  Catalog to provision S3 buckets.
+  :param days_from_deletion: The number of days, from today, that the stack was deleted.
```

Comment:
Maybe the variable should be called `retention_period_days`.  (I *do* like having the units of time, days, in the parameter name.) The description would be "The number of days to retain the stack before deletion."

Reply:
var has been updated